noir/noir-repo/compiler/noirc_evaluator/src/ssa.rs

```diff
@@ -99,10 +105,14 @@ pub struct SsaProgramArtifact {
     pub main_input_witnesses: Vec<Witness>,
     pub main_return_witnesses: Vec<Witness>,
     pub names: Vec<String>,
+    pub error_types: BTreeMap<u64, HirType>,
 }
 
 impl SsaProgramArtifact {
-    fn new(unconstrained_functions: Vec<BrilligBytecode>) -> Self {
+    fn new(
+        unconstrained_functions: Vec<BrilligBytecode>,
+        error_types: BTreeMap<u64, HirType>,
```

Comment:
It would be nice to have a `type ErrorSelector = u64` alias

Reply:
Ah looks like we have this elsewhere.

Reply:
We could probably put the ErrorSelector struct in a more common place, but it uses HirType for constructing itself via hashing, so I'm hesitant on where to put it